src/systemverilog/systemverilog.ts

```diff
@@ -33,9 +33,10 @@ export const conf: IRichLanguageConfiguration = {
 		{ open: '\'', close: '\'' },
 	],
 	folding: {
+		offSide: false,
 		markers: {
-			start: new RegExp("^\\s*#pragma\\s+region\\b"),
-			end: new RegExp("^\\s*#pragma\\s+endregion\\b")
+			start: new RegExp("^\\s*(((initial\\s|always(.*)\\s)?begin)|(case(x|z)?)|(class)|(clocking)|(config)|(covergroup)|(function)|(generate)|(interface)|(module)|(package)|(primitive)|(property)|(program)|(sequence)|(specify)|(table)|(task)\\b)"),
```

Comment:
I think we can simplify this by only looking at `begin` regardless of what comes before it.  That will allow folding on things like if statements but I think that's fine

Reply:
Spencer, I think being able to fold functions and tasks in particular has a lot of value.

maybe we can cut it down to this list?
module, generate, task, function, case, covergroup

Reply:
Just to clarify I was suggesting replaying this `(((initial\\s|always(.*)\\s)?begin` with just `begin` and keep the other parts (so continue to fold on those other things).

Reply:
ok yeah what spencer suggests sounds good to me